GarupaPico/GarupaPico/Model/Song.cs

```diff
@@ -0,0 +1,147 @@
+
+        /// <summary>
+        /// Gets the number of notes of the song.
+        /// </summary>
+        public int SongNotes { get; private set; }
+
+        /// <summary>
+        /// Gets the number of skill keys of the song, fixed at 6 unless game updates.
+        /// </summary>
+        public const int NumOfKeys = 6;
+
+        /// <summary>
+        /// Gets which note triggers skill start, in array (0-5).
+        /// </summary>
+        public int[] Key_start { get; private set; } = new int[NumOfKeys];
+
+        /// <summary>
+        /// Gets which note as the skill ends if lasts 7.0 seconds, in array (0-5).
+        /// </summary>
+        public int[] Key_7_0 { get; private set; } = new int[NumOfKeys];
+
+        /// <summary>
+        /// Gets which note as the skill ends if lasts 7.5 seconds, in array (0-5).
+        /// </summary>
+        public int[] Key_7_5 { get; private set; } = new int[NumOfKeys];
+
+        /// <summary>
+        /// Gets which note as the skill ends if lasts 8.0 seconds, in array (0-5).
+        /// </summary>
+        public int[] Key_8_0 { get; private set; } = new int[NumOfKeys];
+
+        /// <summary>
+        /// Gets which note in the song starts fever.
+        /// </summary>
+        public int FeverStart { get; private set; }
+
+        /// <summary>
+        /// Gets which note in the song as last fever note.
+        /// </summary>
+        public int FeverEnd { get; private set; }
+
+        /// <summary>
+        /// which note triggers skill with high level hitting skill (餡蜜), in array (0-5).
+        /// </summary>
+        public int[] InvertBeat { get; private set; } = new int[NumOfKeys];
+
+        /// <summary>
+        /// Creates dummy song object.
+        /// </summary>
+        public Song() {
+            ID = -1;
+            SongName = "<なし>";
+            Original = false;
+            Band = "N/A";
+            SongLevel = 0;
+            SongNotes = 0;
+            for (int i = 0; i < NumOfKeys; i++) {
+                Key_start[i] = 0;
+                Key_7_0[i] = 0;
+                Key_7_5[i] = 0;
+                Key_8_0[i] = 0;
+                InvertBeat[i] = 0;
+            }
+            FeverStart = 0;
+            FeverEnd = 0;
+        }
+
+        /// <summary>
+        /// Creates normal song object.
+        /// </summary>
+        /// <param name="SongID"></param>
+        /// <param name="SongName"></param>
+        /// <param name="Band"></param>
+        /// <param name="Original"></param>
+        /// <param name="SongLevel"></param>
+        /// <param name="SongNotes"></param>
+        /// <param name="Keys"></param>
+        /// <param name="FeverStart"></param>
+        /// <param name="FeverEnd"></param>
+        /// <param name="InvertBeat"></param>
+        public Song(int ID, string SongName, string Band, bool Original, int SongLevel, int SongNotes, int[][] Keys, int FeverStart, int FeverEnd, int[] InvertBeat) {
```

Comment:
Use Object initializer, not constructor while not necessary.
In this case, you have to provide ALL the variables to initialize the object. This makes the Class very hard to use.